File: src/content/2.4/code/scala/snippet05.scala
implicit def arrowProfunctor = new Profunctor[Function1] {
  def dimap[A, B, C, D](ab: A => B)(cd: C => D)(bc: B => C): A => D =
    cd compose bc compose ab

  def lmap[A, B, C](f: C => A)(ab: A => B): C => B =
    f andThen ab

  def rmap[A, B, C](f: B => C)(ab: A => B): A => C =
    f compose ab
}

File: src/content/2.4/code/ocaml/snippet05.ml
module ProfunctorArrow : Profunctor
with type ('a, 'b) t = 'a -> 'b = struct
  type ('a, 'b) t = 'a -> 'b

  let ( % ) = Fun.compose
  let dimap ab cd f = cd % f % ab
  let lmap f g = g % f
  let rmap f g = f % g
end
